tests/gis_tests/gdal_tests/test_geom.py

```diff
@@ -967,6 +969,60 @@ def test_multi_geometries_m_dimension(self):
                 geom = OGRGeometry(geom_input)
                 self.assertIs(geom.is_measured, True)
 
+    def test_has_curve(self):
+        for geom in self.geometries.curved_geoms:
+            with self.subTest(wkt=geom.wkt):
+                geom = OGRGeometry(geom.wkt)
+                self.assertIs(geom.has_curve, True)
+                msg = f"GEOS does not support {geom.__class__.__qualname__}."
+                with self.assertRaisesMessage(GEOSException, msg):
+                    geom.geos
+        geom = OGRGeometry("POINT (0 1)")
+        self.assertIs(geom.has_curve, False)
+
+    def test_get_linear_and_curve_geometry(self):
+        circular_string = "CIRCULARSTRING (-0.797 0.466,-0.481 0.62,-0.419 0.473)"
+        geom = OGRGeometry(circular_string)
+        linear = geom.get_linear_geometry()
+        self.assertEqual(linear.geom_name, "LINESTRING")
+        self.assertIs(linear.has_curve, False)
```

Comment:
I'm keen to separate these two in individual tests to be a little more aligned with unit testing primitives. What do you think?

Reply:
Sure! Let me have a look at that. Appreciate your feedback 🙏